src/features/user/Settings/ApiKey/ApiKeyForm.tsx

```diff
@@ -37,36 +46,37 @@ export default function ApiKey({}: any) {
   const getApiKey = async () => {
     setIsUploadingData(true);
     try {
-      const res = await getAuthenticatedRequest(
-        '/app/profile/apiKey',
-        token,
-        logoutUser
-      );
+      const res: ApiKeyResponse | ApiCustomError | undefined =
+        await getAuthenticatedRequest('/app/profile/apiKey', token, logoutUser);
       if (res) {
-        setApiKey(res.apiKey);
+        setApiKey(res.apiKey || '');
       }
     } catch (err) {
       setErrors(handleError(err as APIError));
     }
     setIsUploadingData(false);
   };
 
-  const regenerateApiKey = async () => {
+  const regenerateApiKey = async (
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>
+  ) => {
+    e.preventDefault();
     setIsUploadingData(true);
-
     try {
-      const res = await putAuthenticatedRequest(
-        '/app/profile/apiKey',
-        undefined,
-        token,
-        logoutUser
-      );
-      setIsUploadingData(false);
-      setApiKey(res.apiKey);
+      const res: ApiKeyResponse | ApiCustomError | undefined =
+        await putAuthenticatedRequest(
+          '/app/profile/apiKey',
+          undefined,
+          token,
+          logoutUser
+        );
```

Comment:
While this works, `getAuthenticatedRequest` is a generic function and accepts a type.

https://www.typescriptlang.org/docs/handbook/2/generics.html

For an example, refer 

src/features/user/TreeMapper/Analytics/index.tsx Line 20 - 24